Assets/Heart/Modules/Input/Runtime/TouchData.cs

```diff
@@ -1,15 +1,36 @@
 using UnityEngine;
+#if ENABLE_INPUT_SYSTEM
```

Comment:
![image](https://github.com/user-attachments/assets/feb4ace7-e7f7-46cc-b3c4-92f19f638be7)

need add version defines to handle when not install package input-system

#if ENABLE_INPUT_SYSTEM && PANCAKE_INPUTSYSTEM
using UnityEngine.InputSystem;
using UnityEngine.InputSystem.EnhancedTouch;
using Touch = UnityEngine.InputSystem.EnhancedTouch.Touch;
#endif

Reply:
Done

Reply:
I think it okay now

Reply:
class TouchData also needs to receive the same changes as touchWrapper

Reply:
I think we need to redesign based on interface to extend LegacyTouchData and New InputSystem TouchData instead of using #if to handle completely, it looks quite messy and difficult to read and maintain.

Reply:
As for TouchWrapper I think #if the whole class is better than #if each method individually

Reply:
Added interface `ITouchData` and created 2 version for each system.

Checked by install and uninstalling the New Input System. But as I don't have any system in place to check didn't try it out checking if I was getting any error.

Reply:
it looks great, NewInputSystemTouchData is probably a bit too long, what do you think about removing 'new'?